fast64_internal/render_settings.py

```diff
@@ -14,8 +15,73 @@ def on_update_sm64_render_settings(self, context: bpy.types.Context):
-    # TODO: Update render properties from selected OOTLightProperty
-    pass
+    renderSettings: "Fast64RenderSettings_Properties" = context.scene.fast64.renderSettings
+    if renderSettings.ootSceneObject is None or not renderSettings.useObjectRenderPreview:
+        return
+    header = ootGetSceneOrRoomHeader(
+        renderSettings.ootSceneObject,
+        renderSettings.ootSceneHeader,
+        False,
+    )
+    if header is None:
+        return
+    lMode = header.skyboxLighting
+    if lMode == "0x01" or (lMode == "Custom" and not renderSettings.ootForceTimeOfDay):
+        if renderSettings.ootLightIdx >= len(header.lightList):
+            return
+        l = header.lightList[renderSettings.ootLightIdx]
+        renderSettings.ambientColor = tuple(c for c in l.ambient)
+        col0, dir0 = ootGetBaseOrCustomLight(l, 0, False, False)
+        renderSettings.lightColor = tuple(c for c in col0)
+        renderSettings.lightDirection = tuple(d for d in dir0)
+        # col1, dir1 = ootGetBaseOrCustomLight(l, 1, False, False)
+        # renderSettings.light1Color = tuple(c for c in col1)
+        # renderSettings.light1Direction = tuple(d for d in dir1)
+        renderSettings.fogPreviewColor = tuple(c for c in l.fogColor)
+        renderSettings.fogPreviewPosition = (l.fogNear, l.fogFar)
+    else:
+        if header.skyboxLighting == "0x00":
+            tod = header.timeOfDayLights
+            lights = [tod.dawn, tod.day, tod.dusk, tod.night]
+        else:
+            if renderSettings.ootLightIdx + 4 > len(header.lightList):
+                return
+            lights = header.lightList[renderSettings.ootLightIdx:renderSettings.ootLightIdx+4]
+        assert len(lights) == 4
+        todTimes = [0.0, 4.0, 6.0, 8.0, 16.0, 17.0, 19.0, 24.0]
+        todSets  = [  3,   3,   0,   1,    1,    2,    3,    3]
+        t = renderSettings.ootTime
+        for i in range(len(todTimes) - 1):
+            assert t >= todTimes[i]
+            if t < todTimes[i+1]:
+                la, lb = lights[todSets[i]], lights[todSets[i+1]]
+                fade = (t - todTimes[i]) / (todTimes[i+1] - todTimes[i])
+                break
+        else:
+            raise PluginError("OoT time of day out of range")
+        def interpColors(cola, colb, fade):
+            cola = mathutils.Vector(tuple(c for c in cola))
+            colb = mathutils.Vector(tuple(c for c in colb))
+            return cola + (colb - cola) * fade
+        renderSettings.ambientColor = interpColors(la.ambient, lb.ambient, fade)
+        col0a, _ = ootGetBaseOrCustomLight(la, 0, False, False)
+        col0b, _ = ootGetBaseOrCustomLight(lb, 0, False, False)
+        renderSettings.lightColor = col0a + (col0b - col0a) * fade
+        # col1a, _ = ootGetBaseOrCustomLight(la, 1, False, False)
+        # col1b, _ = ootGetBaseOrCustomLight(lb, 1, False, False)
+        # renderSettings.light1Color = col1a * fa + col1b * fb

```

Comment:
same as last suggestion

```suggestion
        # TODO: Implement light1 into shader nodes
        # col1a, _ = ootGetBaseOrCustomLight(la, 1, False, False)
        # col1b, _ = ootGetBaseOrCustomLight(lb, 1, False, False)
        # renderSettings.light1Color = col1a * fa + col1b * fb
```